projects/igniteui-angular/src/lib/directives/size/ig-size.directive.ts

```diff
@@ -0,0 +1,14 @@
+import { Directive, HostBinding, Input } from '@angular/core';
+
+@Directive({
+    selector: '[igSize]',
+})
+export class IgSizeDirective {
+    @Input('igSize')
+    public size: 'small' | 'medium' | 'large' = 'medium';
+
+    @HostBinding('style.--ig-size')
+    public get styleValue(): string {
+      return `var(--ig-size-${this.size})`;
+    }
```

Comment:
Why did you decide to go this route? In my opinion having the input match the name of the getter setter is a lot more straight-forward. Plus, you can still limit what the input type can be via the setter's value type as described in my previous review comments.